fetch/fetchRequest.js

```diff
@@ -0,0 +1,29 @@
+/*!
+ * Copyright (c) 2018-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+require('es6-promise').polyfill();
+require('isomorphic-fetch');
+
+function fetchRequest(method, url, args) {
+  var f = fetch(url, {
+    method: method,
+    headers: args.headers,
```

Comment:
If you are assuming the response is JSON, you should probably inject an 'accepts' here, and if you're sending JSON, you should have 'content-type' set.

Reply:
Correct, the headers should come from here: https://github.com/okta/okta-auth-js/pull/184/files#diff-1c0f1c434b17b7f8795d44a51a14320aR39. 👍

Reply:
This method is generic about the headers, yet we assume it is JSON when it comes to parsing.  I don't have a specific recommendation, but keep your eyes open for a way for the generic bit to not assume something that comes from outside - perhaps a check on the "accepts" header before parsing.

Reply:
Agree, added the check 👍